manual/statements/built/README.md

```diff
@@ -1,5 +1,268 @@
+
+### Inserting data
+
+Insertion of data is straightforward - you're specifying the target table in call to
+`insertInto`, and then provide values to insert either by chaining the several calls to
+`value` function, or by using the `values` function passing the lists or arrays of column
+names & corresponding values. Following 2 examples are equivalent:
+
+```java
+QueryBuilder.insertInto("test").value("id", 4).value("t", "test 4");
+QueryBuilder.insertInto("test").values(Arrays.asList("id", "t"), Arrays.asList(4, "test 4"));
+```
+
+You can also insert JSON-formatted data by calling the `json` function & passing the data:
+
+```java
+QueryBuilder.insertInto("test").json("{\"id\":4, \"t\":\"test 4\"}");
+```
+
+`QueryBuilder` also allows to generate the statement that will use lightweight
+transactions (LWT) to check that inserted data doesn't exist yet.  You just need to add
+the call to `ifNotExists` to the statement:
+
+```java
+QueryBuilder.insertInto("test").value("id", 4).ifNotExists();
+```
+
+It also possible to specify additional metadata for inserted data, such as, TTL or
+timestamp.  This is achieved by using the `using` function and providing the `Using`
+object generated either by `ttl`, or `timestamp` functions of the `QueryBuilder` class. In
+case if you want to specify both of them, you need to wrap other into the call to `and`
+function:
+
+```java
+QueryBuilder.insertInto("test").value("id", 4).using(ttl(10)).and(timestamp(1000));
+```
+
+Besides this, for newer versions of Cassandra it's possible to specify additional
+parameters, such as `DEFAULT UNSET` & `DEFAULT NULL` in the `INSERT INTO ... JSON`
+statements, by using `defaultUnset` & `defaultNull` correspondingly.
+
+### Update statements
+
+Updating the data also relatively straightforward - you specify the data to update,
+condition, and additional options if necessary:
+
+```java
+BuiltStatement updateStatement = QueryBuilder.update("test").with(set("test", 1))
+        .where(eq("id", 1));
+```
+
+Update operation is passed as an argument to the `with` function, and additional
+operations could be chained via `and` calls:
+
+```java
+BuiltStatement updateStatement = QueryBuilder.update("test").with(set("t", "test 1"))
+        .and(set("x", 10)).where(eq("id", 1));
+```
+
+Besides the most often used `set` operation, there is a plenty of operations for work with
+all types of collections (lists, maps & sets): `add`, `discard`, `prepend`, `put`,
+`remove`, `setIdx`, etc.  For full list see the [QueryBuilder]'s documentation.
+
+To update counters you can use the `incr` & `decr` functions that takes the column name,
+and value by which column will be increased or decreased:
+
+```java
+BuiltStatement query = QueryBuilder.update("counters")
+        .with(incr("counter", 1)).where(eq("id", 1));
+```
+
+Similarly to insert statements, it's also possible to perform conditional update by
+calling either `ifExists` to perform only if entry exists, or calling `onlyIf` with
```

Comment:
Suggestion: "...by calling either `ifExists` (to perform the mutation only if the entry exists), or by calling `onlyIf` with a `Clause` object—in this case the row will be updated only if the clause returns true".